scripts/generateTypeSummary.ps1

```diff
@@ -0,0 +1,77 @@
+Param(
```

Comment:
copyright needed, add summary

Reply:
The idea that we compare this generated file with the last generated file to see the actual diff based on compiled dll?

Reply:
Another option we could do is use a disassembler dump and save it. May be a bit messier of a diff.

Reply:
This is more or less what the disassembler is doing, but in a way that we can have cleaner diffs (with sorted types, methods etc.

Reply:
@MIchaelMainer, there will be typeSummary.txt file at the root, and will be updated whenever types are generated. It has a canonical order, so even if the type order changes, the diff will be cleaner. It will also serve as a guide to review the rest of the code files.

A side benefit:
It can also be used to quickly check if we generate a certain type, method or property. There will be single file to search, instead of searching in the file names first, then in the files etc.